deployments/puppet/templates/dotnet.conf.erb

```diff
@@ -0,0 +1,20 @@
+CORECLR_ENABLE_PROFILING=1
+CORECLR_PROFILER={918728DD-259F-4A6A-AC2B-B85E1B658318}
+CORECLR_PROFILER_PATH=/usr/lib/splunk-instrumentation/splunk-otel-dotnet/linux-x64/OpenTelemetry.AutoInstrumentation.Native.so
+DOTNET_ADDITIONAL_DEPS=/usr/lib/splunk-instrumentation/splunk-otel-dotnet/AdditionalDeps
+DOTNET_SHARED_STORE=/usr/lib/splunk-instrumentation/splunk-otel-dotnet/store
+DOTNET_STARTUP_HOOKS=/usr/lib/splunk-instrumentation/splunk-otel-dotnet/net/OpenTelemetry.AutoInstrumentation.StartupHook.dll
+OTEL_DOTNET_AUTO_HOME=/usr/lib/splunk-instrumentation/splunk-otel-dotnet
+OTEL_DOTNET_AUTO_PLUGINS=Splunk.OpenTelemetry.AutoInstrumentation.Plugin,Splunk.OpenTelemetry.AutoInstrumentation
+<% if defined?(@auto_instrumentation_resource_attributes) && @auto_instrumentation_resource_attributes != "" -%>
+OTEL_RESOURCE_ATTRIBUTES=splunk.zc.method=splunk-otel-auto-instrumentation-<%= @auto_instrumentation_version %>,<%= @auto_instrumentation_resource_attributes %>
+<% else -%>
+OTEL_RESOURCE_ATTRIBUTES=splunk.zc.method=splunk-otel-auto-instrumentation-<%= @auto_instrumentation_version %>
+<% end -%>
+<% if defined?(@auto_instrumentation_service_name) && @auto_instrumentation_service_name != "" -%>
+OTEL_SERVICE_NAME=<%= @auto_instrumentation_service_name %>
+<% end -%>
+SPLUNK_PROFILER_ENABLED=<%= @auto_instrumentation_enable_profiler.to_s.downcase %>
+SPLUNK_PROFILER_MEMORY_ENABLED=<%= @auto_instrumentation_enable_profiler_memory.to_s.downcase %>
+SPLUNK_METRICS_ENABLED=<%= @auto_instrumentation_enable_metrics.to_s.downcase %>
+OTEL_EXPORTER_OTLP_ENDPOINT=<%= @auto_instrumentation_otlp_endpoint %>
```

Comment:
If this end point is the gRPC one we need to set `OTEL_EXPORTER_OTLP_PROTOCOL` to `grpc`. The issue is that .NET defaults to `http/protobuf` instead of `grpc`.

Reply:
> If this end point is the gRPC one we need to set `OTEL_EXPORTER_OTLP_PROTOCOL` to `grpc`. The issue is that .NET defaults to `http/protobuf` instead of `grpc`.

Hmm... `libsplunk.so` doesn't support `OTEL_EXPORTER_OTLP_PROTOCOL`: https://github.com/signalfx/splunk-otel-collector/blob/main/instrumentation/src/main.c#L9.